src/pyinterpreter.cpp

```diff
@@ -483,8 +483,32 @@ namespace
             if (error)
             {
                 Py_ssize_t size;
-                auto str = PyUnicode_AsUTF8AndSize(*error, &size);
-                err = std::runtime_error(str ? str : "Unknown python error");
+                std::string str = "";
+                if (traceback == nullptr) {
+                    str = PyUnicode_AsUTF8AndSize(*error, &size);
+                } else {
+                    PyErr_NormalizeException(&type, &value, &traceback);
+                    PyObject *pTraceModule = PyImport_ImportModule("traceback");
+                    if (pTraceModule != nullptr) {
+                        PyObject *pModuleDict = PyModule_GetDict(pTraceModule);
+                        if (pModuleDict != nullptr) {
+                            PyObject *pFunc = PyDict_GetItemString(pModuleDict, "format_exception");
+                            if (pFunc != nullptr) {
+                                PyObject *errList = PyObject_CallFunctionObjArgs(pFunc, type, value,
```

Comment:
PyObject_CallFunctionObjArgs returns a new reference so this is a memory leak here. Please use CPyObject pyFunc = PyObject_CallFunctionObjArgs(...)

Reply:
@hmenyus Done, can you please confirm?